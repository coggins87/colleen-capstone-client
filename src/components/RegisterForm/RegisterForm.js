import React from 'react'
import AuthApiService from '../../services/auth-api-service'

class RegisterForm extends React.Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }
  state = { error: null}
  handleSubmit = e => {
    e.preventDefault()
    const { user_name, password } = e.target

    this.setState({error: null})
    AuthApiService.postUser({
      user_name: user_name.value,
      password: password.value
    })
    .then(user => {
      user_name.value = ''
      password.value = ''
      this.props.onRegistrationSuccess()
    })
    .catch(res => {
      this.setState({error: res.error})
    })
  }
  render(){
    const {error}=this.state
    return(
      <form onSubmit={this.handleSubmit}>
         <div role='alert'>
          {error && <p>{error}</p>}
        </div>
        <label htmlFor='Registration_user_name'>User Name:</label>
        <input required type='text' name='user_name' id='Registration_user_name'></input>
        <label htmlFor='Registration_password'>Password:</label>
        <input required type='text' name='password' id='Registration_password'></input>
        <button>Register</button>
      </form>
    )
  }
}

export default RegisterForm