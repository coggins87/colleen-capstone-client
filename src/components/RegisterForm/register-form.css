._form {
  display: inherit;
  flex-direction: column;
  font-size: 1.0em;
  align-items: center;
}


.form_input, #Registration_user_name {
  padding: 2%;
  margin: 2%;
}

.form_input:hover {
  border: 1px solid rgb(0, 183, 255);
}

.form_button {
  padding: 3%;
  margin-top: 2%;
  border: none;
  font-size: 1em;
  background-color: rgb(0, 183, 255);
  color: white;
}

.form_button:hover{
  background-color: rgb(0, 144, 201);
  color: white;
}

.form_input::placeholder{
  color: rgb(240, 18, 18);
  font-size: 75%;
}

.error {
/*   font-size: 2em;
 */}