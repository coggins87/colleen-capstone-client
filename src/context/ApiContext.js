import React from "react";
import TokenService from "../services/token-service";
import WorkoutService from "../services/workout-search-service";
const ApiContext = React.createContext({
  error: null,
  isLoggedIn: false,
  userId: null,
  setError: () => {},
  clearError: () => {},
  handleLoginSuccess: () => {}
});

export default ApiContext;

export class ApiProvider extends React.Component {
  state = {
    error: null,
    isLoggedIn: false,
    userId: null
  };
  setError = error => {
    console.error(error);
    this.setState({ error });
  };
  clearError = () => {
    this.setState({ error: null });
  };

  onDelete = () => {
    const user_id = this.props.reqs.params.userId;
    WorkoutService.deleteWorkout(user_id);
    this.setState({ workouts: WorkoutService.getUserWorkouts(user_id) });
  };

  nullError = () => {
    this.setState( {error: 'Invalid user, close this window to log out and try again'})
  }

  handleLoginSuccess = () => {
    const read = TokenService.readJwtToken();
    const user_Id = read.user_id;
    this.setState({
      userId: user_Id,
      isLoggedIn: true
    });
  };

  handleLogoutSuccess = () => {
    this.setState({
      userId: null,
      isLoggedIn: false
    });
  };

  isUserLoggedIn = () => {
    if (TokenService.hasAuthToken()) {
      const read = TokenService.readJwtToken();
      const user_Id = read.user_id;
      this.setState({
        userId: user_Id
      });
    }
  };
  render() {
    const value = {
      isUserLoggedIn: this.isUserLoggedIn,
      nullError: this.nullError,
      error: this.state.error,
      isLoggedIn: this.state.isLoggedIn,
      setError: this.setError,
      clearError: this.clearError,
      userId: this.state.userId,
      handleLoginSuccess: this.handleLoginSuccess,
      handleLogoutSuccess: this.handleLogoutSuccess,
      handleDeleteWorkout: this.onDelete
    };
    return (
      <ApiContext.Provider value={value}>
        {this.props.children}
      </ApiContext.Provider>
    );
  }
}
