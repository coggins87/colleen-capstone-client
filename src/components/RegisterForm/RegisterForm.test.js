import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegisterForm from './RegisterForm'
import {ApiProvider} from '../../context/ApiContext'

describe(`RegisterForm component`, () => {
  const props = {
    testObj: {},
    history : {
    testCheck: ()=>{}
    },
  }

  it('renders a RegisterForm by default', () => {
    const wrapper = shallow(<RegisterForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the RegisterForm given props', () => {
    const wrapper = shallow(<RegisterForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the RegisterForm given context', () => {
    const wrapper = shallow(<ApiProvider><RegisterForm /></ApiProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})