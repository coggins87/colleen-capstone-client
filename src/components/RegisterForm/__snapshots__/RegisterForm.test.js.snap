// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component renders a RegisterForm by default 1`] = `
<Fragment>
  <form
    className="_form"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <label
      htmlFor="Registration_user_name"
    >
      User Name:
    </label>
    <input
      className="form_input"
      id="Registration_user_name"
      name="user_name"
      placeholder="Required"
      required={true}
      type="text"
    />
    <label
      htmlFor="Registration_password"
    >
      Password:
    </label>
    <input
      className="form_input"
      id="Registration_password"
      name="password"
      placeholder="Required"
      required={true}
      type="password"
    />
    <label
      htmlFor="Registration_email"
    >
      Email:
    </label>
    <input
      className="form_input"
      id="Registration_email"
      name="email"
      placeholder="Required"
      required={true}
      type="email"
    />
    <button
      className="form_button"
    >
      Register
    </button>
  </form>
  <p
    className="password_rule"
  >
    *Password must be between 8 and 64 characters with at least 1 upper case, lower case, and special character
  </p>
</Fragment>
`;

exports[`RegisterForm component renders the RegisterForm given context 1`] = `
<ContextProvider
  value={
    Object {
      "clearError": [Function],
      "error": null,
      "handleDeleteWorkout": [Function],
      "handleLoginSuccess": [Function],
      "handleLogoutSuccess": [Function],
      "isLoggedIn": false,
      "isUserLoggedIn": [Function],
      "nullError": [Function],
      "setError": [Function],
      "userId": null,
    }
  }
>
  <RegisterForm
    onRegistrationSuccess={[Function]}
  />
</ContextProvider>
`;

exports[`RegisterForm component renders the RegisterForm given props 1`] = `
<Fragment>
  <form
    className="_form"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <label
      htmlFor="Registration_user_name"
    >
      User Name:
    </label>
    <input
      className="form_input"
      id="Registration_user_name"
      name="user_name"
      placeholder="Required"
      required={true}
      type="text"
    />
    <label
      htmlFor="Registration_password"
    >
      Password:
    </label>
    <input
      className="form_input"
      id="Registration_password"
      name="password"
      placeholder="Required"
      required={true}
      type="password"
    />
    <label
      htmlFor="Registration_email"
    >
      Email:
    </label>
    <input
      className="form_input"
      id="Registration_email"
      name="email"
      placeholder="Required"
      required={true}
      type="email"
    />
    <button
      className="form_button"
    >
      Register
    </button>
  </form>
  <p
    className="password_rule"
  >
    *Password must be between 8 and 64 characters with at least 1 upper case, lower case, and special character
  </p>
</Fragment>
`;
